packages/native/src/NavigationContainer.tsx

```diff
@@ -61,7 +61,7 @@ const NavigationContainer = React.forwardRef(function NavigationContainer(
 
   const [isResolved, initialState] = useThenable(getInitialState);
 
-  React.useImperativeHandle(ref, () => refContainer.current);
+  React.useImperativeHandle(ref, () => refContainer.current, [refContainer.current]);
```

Comment:
Refs cannot and shouldn't be used as a dependency since they contain mutable values. The dependency array should only contain immutable data which can trigger a re-render